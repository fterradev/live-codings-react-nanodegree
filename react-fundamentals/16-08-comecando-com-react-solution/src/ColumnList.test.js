import React from 'react';
import { shallow, mount, render } from 'enzyme';
import ColumnList from './ColumnList';

it('renderiza corretamente', () => {
  expect(shallow(<ColumnList />)).toMatchSnapshot()
})

it('renderiza toda a árvore do componente, Done', () => {
  expect(render(<ColumnList title='Done' />)).toMatchSnapshot()
})

it('renderiza toda a árvore do componente, To Do', () => {
  expect(render(<ColumnList title='To Do' />)).toMatchSnapshot()
})

it('ao submeter o formulário, chama addTask', () => {
  const addTask = jest.fn();
  const test = mount(
    <ColumnList
      title='To Do'
      addTask={addTask}
    />
  );
  //console.log(test.find('button').html()) // <button type="submit">Add Task</button>
  test.find('form').simulate('submit')
  expect(addTask).toHaveBeenCalledTimes(1);
})

it('renderiza lista de elementos', () => {
  const test = mount(
    <ColumnList
      title='To Do'
      items={[{ id: 'kjdkjgd', title: 'Meu teste', status: 'To Do' }]}
    />
  );
  expect(test.find('li').length).toBe(1)
})

describe('usuário interage com lista', () => {
  it('usuário seleciona elemento da lista', () => {
    const updateTask = jest.fn();
    const test = mount(
      <ColumnList
        title='To Do'
        items={[{ id: 'kjdkjgd', title: 'Meu teste', status: 'To Do' }]}
        updateTask={updateTask}
      />
    );
    test.find('li').find('input').simulate('change')
    expect(updateTask).toHaveBeenCalledTimes(1)
  })
})

