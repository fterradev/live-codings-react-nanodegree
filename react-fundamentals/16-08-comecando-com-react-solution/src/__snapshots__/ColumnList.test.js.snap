// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza corretamente 1`] = `
<div
  className="column-list"
>
  <h3 />
  <If
    test={false}
  >
    <form>
      <input
        placeholder="Create new task"
        type="text"
      />
      <button
        type="submit"
      >
        Add Task
      </button>
    </form>
  </If>
  <ul
    className="list-items"
  />
</div>
`;

exports[`renderiza toda a árvore do componente, Done 1`] = `
<div
  class="column-list"
>
  <h3>
    Done
  </h3>
  <ul
    class="list-items"
  />
</div>
`;

exports[`renderiza toda a árvore do componente, To Do 1`] = `
<div
  class="column-list"
>
  <h3>
    To Do
  </h3>
  <form>
    <input
      placeholder="Create new task"
      type="text"
    />
    <button
      type="submit"
    >
      Add Task
    </button>
  </form>
  <ul
    class="list-items"
  />
</div>
`;
